e2e/pages/planner.po.ts

```diff
@@ -0,0 +1,104 @@
+import { browser, by, element, ElementFinder, ExpectedConditions } from 'protractor';
+import { Wait } from '../enums/wait';
+import moment = require('moment');
+
+export class PlannerPage {
+    private timelineEvent = by.className('fc-timeline-event');
+    private plannerTitle = element(by.tagName('h2'));
+
+    async getNumberOfVisibleEvents(): Promise<number> {
+        await this.waitForCalendarToLoadEvents();
+        return await element.all(this.timelineEvent).count();
+    }
+
+    private async waitForCalendarToLoadEvents() {
+        await browser
+            .wait(
+                ExpectedConditions.visibilityOf(element.all(this.timelineEvent).first()),
+                Wait.short
+            )
+            .catch(() => Promise.resolve(false));
+    }
+
+    async clickNextButton() {
+        await element(by.className('fc-next-button')).click();
+    }
+
+    async clickPrevButton() {
+        await element(by.className('fc-prev-button')).click();
+    }
+
+    async clickJudgeViewButton() {
+        await element(by.id('planner-judge-view-btn')).click();
+    }
+
+    async clickRoomViewButton() {
+        await element(by.id('planner-room-view-btn')).click();
+    }
+
+    async clickTodayButton() {
+        await element(by.className('fc-today-button')).click();
+    }
+
+    async openDayView() {
+        // Due some reasons events in calendar aren't displayed until some action will be taken
+        // workaround for it is to select already selected view mode (month/week/day/list)
+        await element(by.className('fc-timelineDay-button')).click();
+        await this.clickPrevButton();
+        await this.clickNextButton();
+        await this.waitForCalendarToLoadEvents();
+    }
+
+    async waitUntilVisible() {
+        moment().isoWeekYear();
+        let expectedDate = moment().day(0).format('DD/MM') + ' - ' + moment().day(6).format('DD/MM/YYYY');
+        await browser.wait(ExpectedConditions.visibilityOf(this.plannerTitle), Wait.normal, expectedDate);
+    }
+
+    async getResourceIdByName(nameToSearch: string): Promise<string> {
+        return await element.all(by.className('fc-resource-area'))
+            .last()
+            .all(by.tagName('tr'))
+            .filter(tr => {
+                return tr.element(by.className('fc-cell-text')).getText().then(value => {
+                    return value === nameToSearch;
+                });
+            })
+            .first()
+            .getAttribute('data-resource-id');
+    }
+
+    getRowWithEventsByResource(resourceId: string) {
+        return element.all(by.className('fc-time-area'))
```

Comment:
await is missing here

Reply:
Initially, it was intentional so that the handler to an element could be cloned and reused, but the code for reusing it was removed and changed with a different approach, so, in the end, good catch.